test/test_assert_callback_registration.py

```diff
@@ -282,3 +284,15 @@ def test_failure__wrong_callback(self, automation: MockAutomation, assert_that):
             assert_that(automation) \
                 .registered.run_at(datetime(2019,11,5,22,43,0,0), extra_param='ok') \
                 .with_callback(automation._some_other_function)
+
+    def test_handle(self, automation: MockAutomation, assert_that):
+        automation.enable_register_run_at_during_initialize()
+
+        assert_that(automation) \
+            .registered.run_at(datetime(2019,11,5,22,43,0,0), extra_param='ok') \
+            .with_callback(automation._my_run_at_callback)
+
+        assert len(automation.handles) > 0
+
+        for h in automation.handles:
+            assert h is not None
```

Comment:
Just a general FYI, I was working on fleshing out the time travel and scheduling support in another PR (#20 ). The heavy lifting was done [in scheduler_mocks.py](https://github.com/FlorianKempenich/Appdaemon-Test-Framework/pull/20/files#diff-9c55d9f7ead8f4fc6faf130fa3dc6246). I declined that PR as it got to be pretty massive an unwieldy and am breaking it down in to more manageable sized PRs (see #27 which is the first).

I bring this up because that fully implements the handle return the same was as AppDaemon does it by keeping track of true UUIDs per a registration. I'm not 100% sure what your use case was for this change, but I have feeling that should cover what you were going for. If that looks useful for you it might be worth spending some time reviewing my PR or taking a look at #20 and letting me know if you have any feedback on that.

Reply:
Yes, @lifeisafractal that's a good point.
If for you @foxcris you don't mind to keep running on your fork using the current random int implementation, then maybe we could just close this one and the changes will be implemented by the fully fledged scheduler mocks.

@foxcris Could you just confirm that it would fulfill your use-case?